bin/pulsar-shell

```diff
@@ -18,7 +18,19 @@
 # under the License.
 #
 
-BINDIR=$(dirname "$0")
+# need this for relative symlinks
+PRG="$0"
+while [ -h "$PRG" ] ; do
+  ls=`ls -ld "$PRG"`
+  link=`expr "$ls" : '.*-> \(.*\)$'`
+  if expr "$link" : '/.*' > /dev/null; then
+    PRG="$link"
+  else
+    PRG="`dirname "$PRG"`/$link"
+  fi
+done
```

Comment:
@nicoloboschi may I ask a concrete usage of this change? It seems other scripts all use `BINDIR=$(dirname "$0")`.

Reply:
That's a good question.
This is needed when the binary is a symlink otherwise it won't be able to find the actual pulsar home directory.
For example, if you install the shell with homebrew, a symlink is used. 

To be honest I didn't invented any here, I just copied it from Apache Maven:  https://github.com/apache/maven/blob/f05397a88dc4178fff7ac7d307797cd54db3bef9/apache-maven/src/assembly/shared/mvnvalidate#L1-L13

Reply:
we don't yet. but I already created a draft package for homebrew and it's needed

you can check it out here: https://github.com/nicoloboschi/homebrew-apache-pulsar-shell

Reply:
Looks great :)